back-end/aws-cloud9-api/src/main/java/com/serverless/dal/QuestionDBTable.java

```diff
@@ -0,0 +1,191 @@
+        if(result.size() > 0) {
+        	question = result.get(0);
+        }
+        return question;
+    }
+    
+  
+    
+    public  List<Question> getFormQuestions(String formId) throws IOException {
+    	QuestionDBTable questionTable=new QuestionDBTable();
+    	List<Question> questions = questionTable.list();
+    	List<Question> res = new ArrayList<Question>();
+    	for (Question question : questions)
+        {
+        	if (question.getForm_membership()!=null) {
+            	if(question.getForm_membership().equals(formId))
+            	{
+            		res.add(question);
+            	}
+            	}
+        	
+        }
+
+        return res;
+    }
+    
+    
+    public  List<Question> getRecruiterQuestions(String recruiterId) throws IOException {
+    	QuestionDBTable questionTable=new QuestionDBTable();
+    	List<Question> questions = questionTable.list();
+    	List<Question> res = new ArrayList<Question>();
+    	for (Question question : questions)
+        {
+        	if (question.getRecruiter_id()!=null) {
+            	if(question.getRecruiter_id().equals(recruiterId))
+            	{
+            		res.add(question);
+            	}
+            	}
+        	
+        }
+
+        return res;
+    }
+    
+    
+    
+    public Question update(JsonNode newBody) throws IOException {
+    	logger.info("Call QuestionDBTable::update(" + newBody + ")");
+    	String id = newBody.get("id").asText();
+    	Question question = this.get(id);
+    	if(question == null) {
+    		logger.error("Question with id " + id + " not found");
+    		return null;
+    	}
+    	question.update(newBody);
+    	this.mapper.save(question);
+    	return question;
+    }
+
+    
+
+    
+    
+    
+    public void save(Question newQuestion) throws IOException {
+    	try {
+    	logger.info("Call QuestionDBTable::save(" + newQuestion + ")");
+        this.mapper.save(newQuestion);
+    	}
+    	catch (Exception ex)
+    	{ }
+        
+    }
+    
+
+
+    
+ 
+    
+    public Boolean delete(String id) throws IOException {
+        Question question = null;
+
+        // get product if exists
+        question = this.get(id);
+        if (question != null) {
+          logger.info("Call QuestionDBTable::delete(" + id + ")-> OK");
+          this.mapper.delete(question);
+        } else {
+          logger.warn("Call QuestionDBTable::delete(" + id + ")-> does not exist.");
+          return false;
+        }
+        return true;
+    }
+    
+    
+    public static List<String> singleChars(String s) {
+        String[] answersInTable = s.split("\"");
+        List <String> answer_list= new ArrayList<String>();	
+    	for (int i =0; i< answersInTable.length; i++)
+    	{
+    		if(answersInTable[i].equals("[")== false && answersInTable[i].equals("]")== false && answersInTable[i].equals(",")== false) {
+    		answer_list.add(answersInTable[i]);   }		
+    	}
+    	
+    	return answer_list;
+    }
```

Comment:
Ta funkcja powinna znajdować się w klasie Question i być prywatna, bo tam jest używana, a nie odwołuje się do żadnego pola klasy QuestionDBTable. Poza tym wydaje mi się, że można te odpowiedzi w jsonie przekazać od razu jako tablicę, a nie jako string rozdzielony znakami '\', ale to już musisz się z frontem dogadać. Poza tym nazwa metody nie za bardzo mówi co ona robi.

Reply:
Rozmawiałem z Mateuszem, pytania w jsonie mają być przekazywane jako lista stringów, a nie jako jeden długi string, więc ta funkcja w ogóle nie jest potrzebna